src/sentry/api/paginator.py

```diff
@@ -255,6 +255,68 @@ def get_result(self, limit=100, cursor=None):
         return CursorResult(results=results, next=next_cursor, prev=prev_cursor)
 
 
+class MergingOffsetPaginator(OffsetPaginator):
+    """This paginator uses a function to first look up items from an
+    independently paginated resource to only then fall back to a query set.
+    This is for instance useful if you want to query snuba for the primary
+    sort order and then look up data in postgres.
+    """
+
+    def __init__(
+        self,
+        queryset,
+        data_load_func,
+        apply_to_queryset,
+        key_from_model=None,
+        key_from_data=None,
+        max_limit=MAX_LIMIT,
+        on_results=None,
+    ):
+        OffsetPaginator.__init__(self, queryset, max_limit=max_limit, on_results=on_results)
```

Comment:
`super(OffsetPaginator, self).__init__()` would be the more traditional way to do this in python.

Reply:
The one you pasted is actually incorrect. `super(MergingOffsetPaginator, self)` would be correct. I'm not a huge fan of this pattern.